src/io/GeoJSONReader.cpp

```diff
@@ -0,0 +1,269 @@
+        map[prop.key()] = readProperty(prop.value());
+    }
+    return map;
+}
+
+GeoJSONValue GeoJSONReader::readProperty(nlohmann::json& value) {
+    if (value.is_string()) {
+        return GeoJSONValue::createStringValue(value.get<std::string>());
+    } else if (value.is_number()) {
+        return GeoJSONValue::createNumberValue(value.get<double>());
+    } else if (value.is_boolean()) {
+        return GeoJSONValue::createBooleanValue(value.get<bool>());
+    } else if (value.is_number()) {
+        return GeoJSONValue::createNumberValue(value.get<double>());
+    } else if (value.is_array()) {
+        GeoJSONValue v = GeoJSONValue::createArrayValue();
+        for (auto& el : value.items()) {
+            v.arrayValue.push_back(readProperty(el.value()));
+        }
+        return v;
+    } else if (value.is_object()) {
+        GeoJSONValue v = GeoJSONValue::createObjectValue();
+        for (auto& el : value.items()) {
+            v.objectValue[el.key()] = readProperty(el.value());
+        }
+        return v;
+    } else {
+        return GeoJSONValue::createNullValue();
+    }
+}
+
+std::unique_ptr<geom::Geometry> GeoJSONReader::readFeatureCollectionForGeometry(nlohmann::json& j) {
+    auto featuresJson = j["features"];
+    std::vector<geom::Geometry *>* geometries = new std::vector<geom::Geometry *>();
+    for(auto featureJson : featuresJson) {
+        auto g = readFeatureForGeometry(featureJson);
+        geometries->push_back(g.release());
+    }
+    return std::unique_ptr<geom::GeometryCollection>(geometryFactory.createGeometryCollection(geometries));
+}
+
+GeoJSONFeatureCollection GeoJSONReader::readFeatureCollection(nlohmann::json& j) {
+    auto featuresJson = j["features"];
+    std::vector<GeoJSONFeature> features;
+    for(auto featureJson : featuresJson) {
+        auto f = readFeature(featureJson);
+        features.push_back(f);
+    }
+    return GeoJSONFeatureCollection{features};
+}
+
+
+std::unique_ptr<geom::Geometry> GeoJSONReader::readGeometry(nlohmann::json& j) {
+    std::string type = j["type"];
+    if (type == "Point") {
+        return readPoint(j);
+    } else if (type == "LineString") {
+        return readLineString(j);
+    } else if (type == "Polygon") {
+        return readPolygon(j);
+    } else if (type == "MultiPoint") {
+        return readMultiPoint(j);
+    } else if (type == "MultiLineString") {
+        return readMultiLineString(j);
+    } else if (type == "MultiPolygon") {
+        return readMultiPolygon(j);
+    } else if (type == "GeometryCollection") {
+        return readGeometryCollection(j);
+    }
+    return std::unique_ptr<geom::Geometry>(geometryFactory.createEmptyGeometry());
+}
+
+std::unique_ptr<geom::Point> GeoJSONReader::readPoint(nlohmann::json& j) {
+    std::vector<double> coords = j["coordinates"].get<std::vector<double>>();
+    if (coords.size() == 0) {
+        return std::unique_ptr<geom::Point>(geometryFactory.createPoint(2));
+    } else {
```

Comment:
What if coords.size() == 1?

Reply:
What should we do if there is only one coordinate in the coordinates array?  Return an empty point?

Reply:
I am throwing a ParseException whenever I find a coordinate array with only one value.